### ISSUE TITLE
False positive MissingUseCall on separated usage.

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#9721]
## Steps to Reproduce

```java
package org.eclipse.jgit.api;
public class Git implements AutoCloseable {}
```
```kotlin
inline fun git(repoDir: File, block: Git.() -> Unit) {
	val repo = createGitRepository(repoDir)
	repo.use(block)
}
fun createGitRepository(repoDir: File): Git = TODO()
```

## Expected Behavior
No finding, `.use` is called.

## Observed Behavior

```
createGitRepository doesn't call `use` to access the `Closeable` [MissingUseCall]
``` 

## Context

The sole purpose of the `git` function is to do `.use` correctly and hidden away.

`createGitRepository(repoDir).use(block)` resolves the issue, but it hides the variable from being able to be used in debugger for example.

## Your Environment

* Version of detekt used: 2.0.0-alpha.6
* Version of Gradle used (if applicable): N/A
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: N/A
* Link to your project (if it's a public repository): https://github.com/TWiStErRob/net.twisterrob.gradle/blob/1e8a35e204d86791e3b9dd6c2e4776f34b4eaa34/plugin/versioning/src/testFixtures/kotlin/net/twisterrob/gradle/vcs/git/GitHelpers.kt#L13-L16

**Comment by TWiStErRob** [comment]
Same happened with: https://github.com/TWiStErRob/net.twisterrob.gradle/blob/1e8a35e204d86791e3b9dd6c2e4776f34b4eaa34/plugin/versioning/src/main/kotlin/net/twisterrob/gradle/vcs/git/JGit.kt#L18-L21
```kotlin
internal inline fun <T> Git.walk(block: RevWalk.() -> T): T {
	val walk = RevWalk(this.repository)
	return walk.use(block)
}
```

Note: somehow this is not getting flagged (static function call rather the Kotlin function/constructor call):
```kotlin
internal inline fun <T> inRepo(dir: File, block: Git.() -> T): T {
	val repo = Git.open(dir)
	return repo.use(block)
}
```

**Comment by dzirbel** [comment]
A particularly annoying case:

```
checkNotNull(nullableCloseable()).use { ... }
```

will be flagged because of the indirection from `checkNotNull`/`requireNotNull`. Using `!!` seems to avoid it as a workaround, but it might be worth baking in handling of these wrappers, if full handling of separated usage can't be done.
